### PR TITLE
autodisable curl tests

### PR DESCRIPTION
**Pull request by teto**
- added .editorconfig
- tests: autodisable test in absence of curl

### CODE REVIEW

File: test/auto/db_spec.lua [hunk 1]
```diff
@@ -546,6 +546,10 @@ describe("sqlite.db", function()
   end)
 
   describe(":select", function()
+    if vim.fn.executable("curl") then
+        pending("'curl' program is not available")
```
Comment (tami5) [inline test/auto/db_spec.lua:550]: what's pending 🤔 , part of busted framework?

Comment (teto) [reply test/auto/db_spec.lua:550]: yes. To test you can replace curl with `dadadsadad` to fail the condition and check the output.

**Comment by teto** [comment]
I've copied the .editorconfig from neovim because by default I was adding tabs to the file. I can drop it if you dont like it.

**Comment by tami5** [comment]
> I've copied the .editorconfig from neovim because by default I was adding tabs to the file. I can drop it if you dont like it.

no, that's fine, I don't mind it

**Comment by teto** [comment]
yes that would help ^^. I can always test with a fork but that takes more time.

**Pull request merged**